ITE/6/README.md

```diff
@@ -0,0 +1,574 @@
+define their own custom attestation types if desired.
+
+See [examples](#examples) to get a gist of the schema.
+
+## Design notes
+
+Attestations SHOULD be designed to encourage policies to be "monotonic," meaning
+that deleting an attestation will never turn a DENY decision into an ALLOW. One
+reason for this is because verifiers MUST ignore unrecognized subject digest
+types; if no subject is recognized, the attestation is effectively deleted.
+Example: instead of "deny if a 'has vulnerabilities' attestation exists", prefer
+"deny unless a 'no vulnerabilities' attestation exists".
+
+Attestation designers are free to limit what subject types are valid for a given
+attestation type. For example, suppose a "Gerrit code review" attestation only
+applies to `git_commit` subjects. In that case, a producer of such attestations
+should never use a subject other than `git_commit`.
+
+# Examples
+
+## Provenance example
+
+A provenance-type attestation describing how the
+[curl 7.72.0 source tarballs](https://curl.se/download.html) were built,
+pretending they were built on
+[GitHub Actions](https://github.com/features/actions).
+
+```jsonc
+{
+  // Common attestation fields:
+  "attestation_type": "https://in-toto.io/Provenance/v1",
+  "subject": {
+    "curl-7.72.0.tar.bz2": { "sha256": "ad91970864102a59765e20ce16216efc9d6ad381471f7accceceab7d905703ef" },
+    "curl-7.72.0.tar.gz":  { "sha256": "d4d5899a3868fbb6ae1856c3e55a32ce35913de3956d1973caccd37bd0174fa2" },
+    "curl-7.72.0.tar.xz":  { "sha256": "0ded0808c4d85f2ee0db86980ae610cc9d165e9ca9da466196cc73c346513713" },
+    "curl-7.72.0.zip":     { "sha256": "e363cc5b4e500bfc727106434a2578b38440aa18e105d57576f3d8f2abebf888" }
+  },
+  "materials": {
+    "git+https://github.com/curl/curl@curl-7_72_0": { "git_commit": "9d954e49bce3706a9a2efb119ecd05767f0f2a9e" },
+    "github_hosted_vm:ubuntu-18.04:20210123.1": null,
+    "git+https://github.com/actions/checkout@v2":        { "git_commit": "5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f" },
+    "git+https://github.com/actions/upload-artifact@v2": { "git_commit": "e448a9b857ee2131e752b06002bf0e093c65e571" },
+    "pkg:deb/debian/stunnel4@5.50-3?arch=amd64":               { "sha256": "e1731ae217fcbc64d4c00d707dcead45c828c5f762bcf8cc56d87de511e096fa" },
+    "pkg:deb/debian/python-impacket@0.9.15-5?arch=all":        { "sha256": "71fa2e67376c8bc03429e154628ddd7b196ccf9e79dec7319f9c3a312fd76469" },
+    "pkg:deb/debian/libzstd-dev@1.3.8+dfsg-3?arch=amd64":      { "sha256": "91442b0ae04afc25ab96426761bbdf04b0e3eb286fdfbddb1e704444cb12a625" },
+    "pkg:deb/debian/libbrotli-dev@1.0.7-2+deb10u1?arch=amd64": { "sha256": "05b6e467173c451b6211945de47ac0eda2a3dccb3cc7203e800c633f74de8b4f" }
+  },
+  // Provenance-specific fields:
+  "builder": { "id": "https://github.com/Attestations/GitHubHostedActions@v1" },
+  "recipe": {
+    "type": "https://github.com/Attestations/GitHubActionsWorkflow@v1",
+    "material": "git+https://github.com/curl/curl@curl-7_72_0",
```

Comment:
Is there an implicit or explicit assumption somewhere that these materials that are part of the custom fields also need to be listed as part of the top-level materials? Either way it seems a bit repetitive and error prone, perhaps this can be simplified / normalized with a more structured schema.

Reply:
I think this is addressed in the latest design (`"definedInMaterial: 0"`).